### PR TITLE
Implement MONOMIAL_VEC

### PR DESCRIPTION
**Pull request by lindsayad**
Pretty straightforward synthesis from MONOMIAL and LAGRANGE_VEC. I added an example similar to vector_fe_ex1 but using MONOMIAL_VEC.

### CODE REVIEW

**Comment by lindsayad** [comment]
For @aeslaughter

File: include/fe/fe.h [hunk 1]
```diff
@@ -58,13 +58,19 @@ struct FEOutputType
 template<>
 struct FEOutputType<LAGRANGE_VEC>
 {
-  typedef RealGradient type;
+  typedef RealVectorValue type;
```
Comment (roystgnr) [inline include/fe/fe.h]: This is just for readability, right?  The old code was confusing, not actually broken in some subtle way?

Comment (lindsayad) [reply include/fe/fe.h]: correct

File: src/fe/fe_monomial_vec.C [hunk 2]
```diff
@@ -0,0 +1,845 @@
...
+template <>
+RealGradient
+FE<0, MONOMIAL_VEC>::shape_deriv(const ElemType type,
+                                 const Order order,
+                                 const unsigned int i,
+                                 const unsigned int j,
+                                 const Point & p)
+{
+  Real value = FE<0, MONOMIAL>::shape_deriv(type, order, i, j, p);
+  return libMesh::RealGradient(value);
```
Comment (roystgnr) [inline src/fe/fe_monomial_vec.C]: Ha!  But here you're back to writing RealGradient again.  The confusion will never be eradicated.

Comment (lindsayad) [reply src/fe/fe_monomial_vec.C]: Haha I will change it!

File: examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    libMesh::out << " " << argv[i];
+
+  libMesh::out << std::endl << std::endl;
+
+  // Skip this 2D example if libMesh was compiled as 1D-only.
+  libmesh_example_requires(2 <= LIBMESH_DIM, "2D support");
+
+  // Create a mesh, with dimension to be overridden later, on the
+  // default MPI communicator.
+  DistributedMesh mesh(init.comm());
```
Comment (roystgnr) [inline examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C]: Do we want to force DistributedMesh here?

**Comment by roystgnr** [comment]
:+1:, but I'm confused about a couple bits:

Why are you using a nonlinear solver here?  Problem looks linear to me.

You mean "match LAGRANGE_VEC version" loosely, right?  There's no weird isomorphism between the C0 P2 and DG P1 approximations?

File: examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C (hunk 3 above)
Comment (lindsayad) [reply examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C]: I suppose not

**Comment by lindsayad** [comment]
> Why are you using a nonlinear solver here? Problem looks linear to me.

It is linear, but

- A non-linear configuration is what I'm most used to
- It seems like there's a lot of linear examples in libMesh, and... 

> You mean "match LAGRANGE_VEC version" loosely, right? There's no weird isomorphism between the C0 P2 and DG P1 approximations?

Yes I do mean loosely, as in the approximate solutions both approach the same true solution. Also since the `LAGRANGE_VEC` example uses a linear solver, I thought it might be a nice demo to show solving the same problem not only with a different basis but with a different solver.

File: examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C (hunk 3 above)
Comment (lindsayad) [reply examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C]: Changed to `Mesh`

**Pull request merged**